tests/unit_tests/client/address_book/storage.cc

```diff
@@ -0,0 +1,196 @@
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,          //
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF    //
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.               //
+ */
+
+#define BOOST_TEST_DYN_LINK
+
+#include <boost/test/unit_test.hpp>
+
+#include <iostream>
+
+#include "client/address_book/storage.h"
+
+#include "core/router/context.h"
+
+struct AddressBookStorageFixture
+{
+  /// @alias Subscription
+  /// @brief Subscription alias for readability and convenience
+  using Subscription = kovri::client::AddressBook::Subscription;
+
+  AddressBookStorageFixture() : m_Storage(nullptr)
+  {
+    // Setup temporary data directory
+    kovri::core::context.SetCustomDataDir(temp_path.string());
+
+    // Ensure Client and AddressBook paths exist
+    kovri::core::EnsurePath(
+        kovri::core::GetPath(kovri::core::Path::Client));
+
+    kovri::core::EnsurePath(
+        kovri::core::GetPath(kovri::core::Path::AddressBook));
+
+    // Setup storage
+    m_Storage = GetStorageInstance();
+  }
+
+  ~AddressBookStorageFixture()
+  {
+    RemoveFiles();
+  }
+
+  /// @brief Remove files created by AddressBookStorage
+  void RemoveFiles() try
+    {
+      boost::filesystem::directory_iterator end_itr;
+      for (boost::filesystem::directory_iterator dir_itr(
+               kovri::core::GetPath(kovri::core::Path::AddressBook));
+           dir_itr != end_itr;
+           ++dir_itr)
+        {
+          auto temp_file = dir_itr->path().filename();
+
+          // Only remove files created by AddressBookStorage
+          if (!boost::starts_with(temp_file.string(), test_prefix))
+            boost::filesystem::remove(temp_file);
+        }
+    }
+  catch (boost::filesystem::filesystem_error& er)
+    {
+      std::cerr << __func__ << ": " << er.what();
+    }
+
+  /// @brief Insert test hosts into an address map
+  void ToAddressMap(Subscription source)
+  {
+    for (const auto& host : m_Hosts)
+      {
+        const std::size_t pos = host.find("=");
+        const std::string name = host.substr(0, pos);
+        const std::string address = host.substr(pos);
+        kovri::core::IdentityEx ident;
+        ident.FromBase64(address);
+        m_Addresses[name] = std::make_pair(ident.GetIdentHash(), source);
+      }
+  }
+
+  /// @brief Create new storage instance
+  /// @notes Called after creating custom data directory
+  std::unique_ptr<kovri::client::AddressBookStorage> GetStorageInstance()
+  {
+    return std::make_unique<kovri::client::AddressBookStorage>();
+  }
+
+
+  std::string const test_prefix{"addr_test_"};  // used for temporary test files
+  kovri::client::AddressBook::AddressMap m_Addresses;
+  std::unique_ptr<kovri::client::AddressBookStorage> m_Storage;
+  boost::filesystem::path temp_path = boost::filesystem::temp_directory_path();
```

Comment:
A security concern but these are for tests so that shouldn't be an issue.

Reply:
Security concern because of leaking potentially private data, or is there a filesystem integrity issue?